cli/pcluster/cli_commands/delete.py

```diff
@@ -33,6 +35,7 @@ def delete(args):
                 )
             )
         utils.warn("Cluster {0} has already been deleted.".format(args.cluster_name))
+        _terminate_cluster_nodes(stack_name)
```

Comment:
Why are we terminating nodes here if the stack does not exist?

Reply:
Consider this case:
- user runs a pcluster delete and the CustomResource fails for any reason to terminate some or all EC2 instances. Stack is set to DELETE_FAILED but the Lambda function for the cleanup is removed and this can't be avoided
- user runs again a pcluster delete or deletes the stack from the console and the stack completes successfully because there is no lambda function handling the cleanup

Because of this it is unlikely but possible to end up in situations where the stack does not exist but capacity for that cluster is running. pcluster delete at that point can be used to perform a clean-up

Reply:
OK. I'm wondering if the user couldn't get confused by seeing the above message and the ones coming from `_terminate_cluster_nodes` when running the command against a wrong - non existent cluster name. It's a very minor issue I know; maybe we could just print an additional message informing that we are trying a clean up in any case, or only print messages only when we are really terminating instances.
Also a FIXME comment to address this would be OK if we are short in time.

Reply:
Slightly rephrased and added a FIXME